code/parse/sexp.cpp

```diff
@@ -16514,22 +16561,31 @@ void sexp_set_debris_field(int n)
 		}
 	}
 
-	int debris1 = -1;
-	if (n >= 0) {
-		debris1 = get_asteroid_index(CTEXT(n));
-		n = CDR(n);
-	}
+	SCP_vector<int> debris_types;
+	if (!new_sexp) {
+		if (n >= 0) {
+			int debris1 = get_asteroid_index(CTEXT(n));
+			if (debris1 > 0) {
```

Comment:
Other places using this function check for ``>= 0`` instead, should this be here as well?

Reply:
Ah yes, good catch. Fixed.